### PR TITLE
Backmerge: Market List

### PR DESCRIPTION
**Pull request by mmvsha**
Includes: 
- Changes for MarketList component and styles, api/rest.js and utils/restFetch.js

### CODE REVIEW

File: src/utils/fetch.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+export const requestFromRestAPIPage = async (endpoint, queryparams = {}, page = 0, pageSize = 100) =>
+  requestFromRestAPI(endpoint, {
+    ...queryparams,
+    limit: pageSize,
+    size: pageSize,
+    offset: page * pageSize,
+  })
+
+export const requestFromRestAPIAllPages = async (endpoint, queryparams, pageSize, limit = 100000) => {
```
Comment (apanizo) [inline src/utils/fetch.js]: This method should contain tests. At least: 
* Fetching one endpoint having 0 results
* Fetching one endpoint having 1 page 
* Fetching one endpoint having 2.7 pages (meaning the last one is not complete), fetching all pages.
* If you feel inspired and you want to add a stress test in order to know the limits, even better.

File: src/components/MarketList/index.js [hunk 2]
```diff
@@ -243,28 +248,27 @@ class MarketList extends Component {
...
-        <div className="marketListPage__markets">
-          <div className="container">
-            <div className="row">
-              {this.renderMarkets()}
-              {this.renderMarketFilter()}
+          )}
+          <div className="marketListPage__markets">
+            <div className="container">
+              <div className="row">
+                {this.renderMarkets()}
```
Comment (apanizo) [inline src/components/MarketList/index.js]: I do not know if you are refactoring now but those functions (```this.renderMarkets() and this.renderMarketFilter()```) should be private components.

Comment (mmvsha) [reply src/components/MarketList/index.js]: @apanizo We will do it after the backmerge is complete

**Comment by andre-meyer** [comment]
I suggest we don't refactor until we finished backmerging. Otherwise there is a high chance for things we miss, or error we introduce. Let's refactor after the backmerge is completely done, like we decided in the Proposal.

File: src/components/MarketList/index.js (hunk 2 above)
Comment (apanizo) [reply src/components/MarketList/index.js]: Sure, no problem.

**Comment by apanizo** [comment]
@andre-meyer sure, no problem. But tests should be done now, right? I mean, those ones for ```requestFromRestAPIAllPages```, right?

**Comment by andre-meyer** [comment]
Please let's do them in the next step. We will do them, I promise, I just don't want the backmerging to take even longer

**Comment by apanizo** [comment]
Sure! Thanks for clarifying.

File: src/components/MarketMySharesForm/index.js [hunk 3]
```diff
@@ -444,7 +444,7 @@ MarketMySharesForm.propTypes = {
   market: marketShape,
   selectedSellAmount: PropTypes.string,
   limitMargin: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-  marketShares: PropTypes.arrayOf(PropTypes.object),
```
Comment (andre-meyer) [inline src/components/MarketMySharesForm/index.js]: I think we have a marketShares Proptypes, if you want you can change it here to be `PropTypes.arrayOf(marketSharePropType)`

Comment (andre-meyer) [reply src/components/MarketMySharesForm/index.js]: Oops, I mean `marketShares: PropTypes.objectOf(marketSharePropType),`.

**Pull request merged**